File: 455.py


class Solution:
    def findContentChildren(self, g: list, s: list) -> int:
        g.sort()
        s.sort()
        p1 = 0
        p2 = 0
        res = 0
        while p1 < g.__len__():
            while p2 < s.__len__() and s[p2] < g[p1]:
                p2 += 1

            if p2 == s.__len__():
                return res
            else:
                res += 1
                p1 += 1
                p2 += 1

        return res

File: 455.cpp
// class Solution {
// public:
//     int findContentChildren(vector<int>& g, vector<int>& s) 
//     {
//         sort(g.begin(), g.end());
//         sort(s.begin(), s.end());

//         int p1 =0;
//         int p2=0;
//         int res=0;

//         while(p1<g.size())
//         {
//             while(p2<s.size() && s[p2]<g[p1])
//             {
//                 p2++;
//             }
//             if(p2==s.size())
//                 return res;
//             else
//             {
//                 res++;
//                 p1++;
//                 p2++;
//             }

//         }
//         return res;
        
//     }
// };